interfaces/input_parameters/ring.py

```diff
@@ -182,182 +130,348 @@ class Ring:
 
     """
 
-    def __init__(self, ring_length, alpha, Particle,
-                 momentum=None, kin_energy=None, energy=None,
-                 bending_field=None, bending_radius=None,
-                 **kwargs):
-
-        # Checking that at least one synchronous data input is passed
-        syncDataTypes = ('momentum', 'kin_energy', 'energy', 'B_field')
-        syncDataInput = (momentum, kin_energy, energy, bending_field)
-        assrt.single_not_none(*syncDataInput,
-                              msg='Exactly one of '+str(syncDataTypes) +
-                              ' must be declared',
-                              exception=excpt.InputError)
-
-        # Checking that the bending_radius is passed with the bending_field
-        if bending_field is not None and bending_radius is None:
-            raise excpt.InputError("If bending_field is used, bending_radius "
-                                   + "must be defined.")
-
-        # Taking the first synchronous_data input not declared as None
-        # The assertion above ensures that only one is declared
-        for func_type, synchronous_data in zip(syncDataTypes, syncDataInput):
-            if synchronous_data is not None:
-                break
-
-        # Setting ring length, circumerence, radius, bending radius if defined
-        self.ring_length = np.array(ring_length, ndmin=1, dtype=float)
-        self.ring_circumference = np.sum(self.ring_length)
-        self.ring_radius = self.ring_circumference/(2*np.pi)
-
-        if bending_radius is not None:
-            self.bending_radius = float(bending_radius)
-        else:
-            self.bending_radius = bending_radius
+    def __init__(self, Particle, Section_list, **kwargs):
 
         # Primary particle mass and charge used for energy calculations
         # If a string is passed, will generate the relevant Particle object
         # based on the name
         if isinstance(Particle, beam.Particle):
             self.Particle = Particle
-        else:
+        elif isinstance(Particle, str):
             self.Particle = beam.make_particle(Particle)
-
-        # Reshaping the input synchronous data to the adequate format and
-        # get back the momentum program from RingOptions
-        if not isinstance(synchronous_data, dTypes._ring_program):
-            synchronous_data = \
-                dTypes._ring_program.conversions[func_type](synchronous_data)
-
-        if synchronous_data.shape[0] != len(self.ring_length):
-            raise excpt.InputDataError("ERROR in Ring: Number of sections " +
-                                       "and ring length size do not match!")
-
+        else:
+            raise excpt.InputError("The input Particle should be a " +
+                                   "Particle object or a str.")
```

Comment:
[make_particle](https://github.com/scpalbright/blond_common/blob/master/interfaces/beam/beam.py#L54) accepts str or dict with 'user_mass' and 'user_charge' as keys.  There's an internal check that it's one of the two so the type check here is not  needed.
```suggestion
        elif isinstance(Particle, (str, dict)):
            self.Particle = beam.make_particle(Particle)
        else:
                        raise excpt.InputError("The input Particle should be a " +
                                   "Particle object, str or dict.")
```

Reply:
Corrected, to be seen if we try to catch the exception but I think it's ok if all is handled by make_particle